.github/workflows/nightly.yml

```diff
@@ -0,0 +1,52 @@
+name: Deploy Nightly
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '8 20 * * *' # Midnight-ish NA time. Offset from the hour to hopefully avoid job delays
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: 'yarn'
+      - name: Set git config
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+      - name: Yarn install
+        run: |
+          yarn install --immutable --immutable-cache
+      - name: Setup .env
+        run: |
+          printf '%s' "$ENVFILE" > apps/frontend/.env.local
+        env:
+          ENVFILE: ${{ secrets.ENVFILE }}
+      - name: Build genshin-optimizer
+        run: |
+          NX_URL_GITHUB_GO_CURRENT_VERSION="https://github.com/${{ github.repository }}/commit/$(git rev-parse HEAD)" \
```

Comment:
Maybe consider changing the package.json version to have the current date postpended as well, to indicate its a nightly